epimodel/tasks.py

```diff
@@ -0,0 +1,392 @@
+        start_date = (
+            utils.utc_date(self.start_date) if self.start_date else d.get_start_date()
+        )
+        log.info(f"Generating scenarios with start_date {start_date.ctime()} ...")
+        batch_module.generate_simulations(
+            b,
+            d,
+            est,
+            rds=rds,
+            config=ConfigYaml.load(self.input()["config_yaml"].path),
+            start_date=start_date,
+            top=self.top,
+        )
+        log.info(f"Generated batch {b.path!r}:\n  {b.stats()}")
+        b.close()
+
+        # todo: not sure about this
+        # if "invoked_by_subcommand" in ctx.parent.__dict__:
+        #    ctx.parent.batch_file = b.path
+
+
+@inherits(RegionsDatasetTask, GenerateGleamBatch)
+class ExportGleamBatch(luigi.Task):
+    exports_dir = luigi.Parameter(default="~/GLEAMviz/data/sims/")
+    overwrite = luigi.BoolParameter(default=False)
+
+    def run(self):
+        batch_file = self.input()["batch_file"]
+        batch = Batch.open(batch_file)
+        gdir = Path(self.exports_dir)
+        log.info(
+            f"Creating GLEAM XML definitions for batch {self.batch_file} in dir {gdir} ..."
+        )
+        batch.export_definitions_to_gleam(
+            gdir.expanduser(), overwrite=self.overwrite, info_level="INFO"
+        )
+
+    def requires(self):
+        return {
+            "batch_file": self.clone(GenerateGleamBatch),
+            "region_dataset": self.clone(RegionsDatasetTask),
+        }
+
+    def output(self):
+        # TODO: this needs to be improved and actually catch the generated files and not just the
+        # default directory
+        return luigi.LocalTarget(self.exports_dir)
+
+
+@requires(ExportGleamBatch)
+class GleamvizResults(luigi.ExternalTask):
+    """This is done manually by a user via Gleam software"""
+
+    # I expect that this is something like "~/GLEAMviz/data/sims/some-batch-file.hdf5"
+    gleamviz_result = luigi.Parameter()
+
+    def output(self):
+        return luigi.LocalTarget(self.gleamviz_result)
+
+
+@inherits(GleamvizResults, RegionsDatasetTask, ConfigYaml)
+class ImportGleamBatch(luigi.Task):
+    allow_missing = luigi.BoolParameter(default=True)
+    result_batch_file_path = luigi.Parameter(default="data/result-batch-file.hdf5")
+
+    def requires(self):
+        return {
+            "gleamviz_result": self.clone(GleamvizResults),
+            "region_dataset": self.clone(RegionsDatasetTask),
+            "config_yaml": self.clone(ConfigYaml),
+        }
+
+    def run(self):
+        batch_file = self.input()["gleamviz_result"].path
+        # assuming batch file is in the gleamviz results dir
+        simulation_directory = os.path.dirname(batch_file)
+
+        config_yaml = ConfigYaml.load(self.input()["config_yaml"].path)
+        regions_dataset = RegionsDatasetTask.load_dilled_rds(
+            self.input()["regions_dataset"].path
+        )
+
+        # copy the gleamviz results into a temporary directory
```

Comment:
this IMO solves the problem with overwriting. I copy the data into a temporary location and work on that, later move it to the final destination.